simd/meson.build

```diff
@@ -0,0 +1,29 @@
+incs += include_directories('.')
+
+libp2p_src += files(
+    'cpuinfo_x86.cpp',
+    'p2p_simd.cpp',
+    'p2p_sse41.cpp',
+)
+
+libp2p_headers += files(
+    'cpuinfo_x86.h',
+    'p2p_simd.h',
+)
+
+cpp_features = ['-DP2P_SIMD']
+
+if cpp.get_argument_syntax() == 'gcc'
+    cpp_features += [
+        '-msse4.1',
+    ]
+elif cpp.get_argument_syntax() == 'msvc'
+    cpp_features += [
+        '/arch:AVX2'
+    ]
+endif
```

Comment:
I'm not sure if I should add a exception for clang-cl like this one: https://github.com/moi15moi/libp2p/blob/f52c14efc88c53317d00f45610a9e42030ba8a21/_msvc/p2p_sources/p2p_sources.vcxitems#L30